sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -129,56 +131,79 @@ inline void piCheckResult(PiResult pi_result) {
   piCheckThrow<cl::sycl::runtime_error>(pi_result);
 }
 
-template <typename FnType> class Trace {
+#define PI_TRACE_ENABLED (std::getenv("SYCL_PI_TRACE") != nullptr)
+
+template <typename FnType, size_t FnOffset> class Trace {
 private:
   FnType m_FnPtr;
-  static bool m_TraceEnabled;
+  std::string m_FnName;
 
 public:
-  Trace(FnType FnPtr, const std::string &FnName) : m_FnPtr(FnPtr) {
-    if (m_TraceEnabled)
-      std::cout << "---> " << FnName << "(";
-  }
-
+  Trace();
   template <typename... Args> PiResult operator()(Args... args) {
-    if (m_TraceEnabled)
+    if (PI_TRACE_ENABLED) {
+      std::cout << "---> " << m_FnName << "(";
       printArgs(args...);
+    }
 
-    auto r = m_FnPtr(args...);
+    PiResult r = m_FnPtr(args...);
 
-    if (m_TraceEnabled) {
+    if (PI_TRACE_ENABLED) {
       std::cout << ") ---> ";
       std::cout << (print(r), "") << std::endl;
     }
     return r;
```

Comment:
So, +2 `getenv` calls on **each** PI API call, right? I guess this is some kind of overkill and definitely will affect performance